functions/Get-DbaTempdbUsage.ps1

```diff
@@ -0,0 +1,85 @@
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+    .LINK
+    https://dbatools.io/Get-DbaTempdbUsage
+    .EXAMPLE
+    Get-DbaTempdbUsage -SqlInstance localhost\SQLDEV2K14
+	
+	Gets tempdb usage for localhost\SQLDEV2K14
+    .EXAMPLE
+    Get-DbaTempdbUsage -SqlInstance localhost\SQLDEV2K14 -Detailed
+	
+	Gets detailed tempdb usage for localhost\SQLDEV2K14
+    #>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[object[]]$SqlInstance,
+		[object]$SqlCredential,
+		[switch]$Detailed,
+		[switch]$Silent
+	)
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlInstance)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $instance -SqlCredential $sqlcredential
+			}
+			catch
+			{
+				Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $Instance
+			}
+			
+			if ($server.VersionMajor -le 9)
+			{
+				Stop-Function -Message "This function is only supported in SQL Server 2008 or higher."
+				continue
+			}
+			
+			if ($Detailed -eq $true)
+			{
+				$sql = "SELECT SERVERPROPERTY('MachineName') AS ComputerName, SERVERPROPERTY('InstanceName') AS InstanceName, SERVERPROPERTY('ServerName') AS SqlInstance, t.session_id AS SPID, r.command AS StatementCommand, r.start_time AS StartTime, t.user_objects_alloc_page_count * 8 AS UserObjectAllocatedSpace, t.user_objects_dealloc_page_count * 8 AS UserObjectDeallocatedSpace, t.internal_objects_alloc_page_count * 8 AS InternalObjectAllocatedSpace, t.internal_objects_dealloc_page_count * 8 AS InternalObjectDeallocatedSpace, r.reads AS RequestedReads, r.writes AS RequestedWrites, r.logical_reads AS RequestedLogicalReads, r.cpu_time AS RequestedCPUTime, s.is_user_process AS IsUserProcess, s.[status] AS Status, DB_NAME(s.database_id) AS [Database], s.login_name AS LoginName, s.original_login_name AS OriginalLoginName, s.nt_domain AS NTDomain, s.nt_user_name AS NTUserName, s.[host_name] AS HostName, s.[program_name] AS ProgramName, s.login_time AS LoginTime, s.last_request_start_time AS LastRequestedStartTime, s.last_request_end_time AS LastRequestedEndTime FROM sys.dm_db_session_space_usage AS t INNER JOIN sys.dm_exec_sessions AS s ON s.session_id = t.session_id INNER JOIN sys.dm_exec_requests AS r ON r.session_id = s.session_id WHERE t.user_objects_alloc_page_count + t.user_objects_dealloc_page_count + t.internal_objects_alloc_page_count + t.internal_objects_dealloc_page_count > 0"
+				$server.ConnectionContext.ExecuteWithResults($sql).Tables
```

Comment:
So build the $sql variable first, outside of the if/else blocks, and then use a Select-Object based on the -Detailed switch, is it?